script/MoveMediaManager.js

```diff
@@ -0,0 +1,165 @@
+            appendTo: '.dokuwiki',
+            modal: true,
+            close: function () {
+                // do not reuse the dialog
+                // https://stackoverflow.com/a/2864783
+                jQuery(this).dialog('destroy').remove();
+            }
+        });
+    }
+
+    /**
+     * Create the form for the old and new file names
+     *
+     * @param {string} src
+     * @returns {HTMLDivElement}
+     */
+    #buildForm(src) {
+        const wrapper = document.createElement('div');
+        const form = document.createElement('form');
+        wrapper.appendChild(form);
+
+        const intro = document.createElement('p');
+        intro.innerText = LANG.plugins.move.dialogIntro;
+        form.appendChild(intro);
+
+        const original = document.createElement('input');
+        original.type = 'hidden';
+        original.name = 'move-old-filename';
+        original.value = src;
+        form.appendChild(original);
+
+        // strip file extension and put it in a readonly field so it may not be modified
+        const fileExt = document.createElement('input');
+        fileExt.type = 'text';
+        fileExt.readOnly = true;
+        fileExt.size = 5;
+        fileExt.name = 'move-file-ext';
+        fileExt.value = src.split('.').pop();
+
+        const destination = document.createElement('input');
+        destination.type = 'text';
+        destination.className = 'edit';
+        destination.name = 'move-new-filename';
+        destination.value = src.substring(0, src.length - (fileExt.value.length + 1));
+        destination.size = 50;
+        form.appendChild(destination);
+        form.appendChild(fileExt);
+
+        const button = document.createElement('button');
+        button.innerText = LANG.plugins.move.moveButton;
+        form.appendChild(button);
+
+        form.addEventListener('submit', this.#requestMove.bind(this, form));
+
+        return wrapper;
+    }
+
+    /**
+     * Send move request to backend
+     *
+     * @param {HTMLFormElement} form
+     * @param {Event} event
+     * @returns {Promise<void>}
+     */
+    async #requestMove(form, event) {
+
+        event.preventDefault();
+        event.stopPropagation();
+
+        const src = form.querySelector('input[name="move-old-filename"]').value;
+        const dst = form.querySelector('input[name="move-new-filename"]').value;
+        const ext = form.querySelector('input[name="move-file-ext"]').value;
+
+        jQuery.post(
+            DOKU_BASE + 'lib/exe/ajax.php',
+            {
+                call: 'plugin_move_rename_mediamanager',
+                src: src,
+                dst: dst + '.'  + ext
+            },
+            // redirect or display error
+            function (result) {
+                if (result.error) {
+                    const errorText = document.createElement('strong');
+                    errorText.innerText = result.error.join();
+                    errorText.style = 'color: red;';
```

Comment:
I think it would be nicer to re-use the message box style that is usually used in DokuWiki for displaying error messages (if possible, I haven't checked if there is anything in it that wouldn't work in this dialog).

Reply:
Custom style replaced with DokuWiki's typical `<div class="error">`